pandas/core/generic.py

```diff
@@ -278,7 +278,7 @@ def _setup_axes(cls, axes, info_axis=None, stat_axis=None, aliases=None,
         if build_axes:
 
             def set_axis(a, i):
-                setattr(cls, a, properties.AxisProperty(i))
+                setattr(cls, a, properties.AxisProperty(i, a))
```

Comment:
This `setattr` has always bothered me.  It kind of makes since in a general-case world with `PanelND`, but if we're only interested in `Series/DataFrame/[barely-]Panel` wouldn't it be clearer just to write:

```
class DataFrame(NDFrame):
    index = properties.AxisProperty(0)
    columns = properties.AxisProperty(1)
```
?

Reply:
Yep, fully agree that if we only have 1D/2D this makes much more sense. But maybe we should only clean that up when Panel is removed?

Reply:
Yep, fully agree that if we only have 1D/2D this makes much more sense. But maybe we should only clean that up when Panel is removed?